File: test-react.py
import asyncio
from playwright.async_api import async_playwright

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page()
        
        page.on("console", lambda msg: print(f"BROWSER CONSOLE: {msg.text}"))
        page.on("pageerror", lambda err: print(f"BROWSER ERROR: {err}"))
        
        try:
            await page.goto("http://localhost:5173", wait_until="networkidle")
            print("Page loaded successfully.")
        except Exception as e:
            print(f"Navigation error: {e}")
            
        await browser.close()

asyncio.run(main())

File: test-react.mjs
import puppeteer from 'puppeteer';

(async () => {
  const browser = await puppeteer.launch({ headless: 'new' });
  const page = await browser.newPage();
  
  page.on('console', msg => console.log('BROWSER CONSOLE:', msg.text()));
  page.on('pageerror', err => console.log('BROWSER ERROR:', err.toString()));
  
  try {
    await page.goto('http://localhost:5173', { waitUntil: 'networkidle0', timeout: 5000 });
    console.log("Page loaded successfully.");
  } catch (err) {
    console.log("Navigation error:", err.message);
  }
  
  await browser.close();
  process.exit(0);
})();
